modules/references/client/api/references.api.js

```diff
@@ -76,6 +76,24 @@ export async function read({ userFrom, userTo }) {
   );
 }
 
+/**
+ * API request: read references written by loggedIn user, filter them by userTo,
+ * and sort by 'created' field starting from the most recent date
+ *
+ * @param {string} userTo - id of user who received the reference
+ * @returns Promise<Reference[]> - array of the found references
+ */
+export async function readMine({ userTo }) {
+  const params = { userTo: userTo };
+  const { data: reference } = await axios.get('/api/my-reference', {
+    params,
+  });
+
+  return reference !== null
```

Comment:
Should we return 404 here rather than 200 and `null`? Seems like returning 404 is a normal thing to do when something is not found. On the other hand, reference not being found here is a part of the normal flow, not an exceptional situation.

Reply:
I think 404 is a normal behavior in REST API principles (although I know our APIs are more of a mixture, especially when it comes to routes). We have a few APIs already doing the same.

Then you just need to take better care to `catch()` it at the client-side so that you don't bubble up errors.

Not a strong opinion tho, I'd leave this up to you :-)

Reply:
Another way to look at it that might help you decide:

If you're fetching for an object, lack of the object should be 404.

If you're fetching an answer to a question, then 200 with yes/no seems appropriate.

Does that make sense?

Reply:
Yeah, absolutely, everything makes sense. Here the question is not what the endpoint should return (I agree that 404 is more suitable) but rather whether this endpoint is the right thing to call when we just need to ask this yes/no question

Reply:
I am thinking again about it, now having one route back again looks pretty good to me :D

`/api/experiences-by/:experienceUserId?userTo=<userTo>&includeReplies=<includeReplies>`

**Checking if haven't shared experience yet:**
`/api/experiences-by/:experienceUserId?userTo=<userTo>&include_replies=false`
status 200 and `len(fetched_experiences) == 0`

** Displaying on the profile page: **
`/api/experiences-by/:experienceUserId?include_replies=true`

Reply:
I think this has two aspects:

- What makes more sense as an API route?
- How does the underlying code look like?

I agree it can make sense as a single route for all this, but at the same time does it make the controller for that route more complex and harder to reason about, when it has to deal with:
- Returning replies or not (instead of just defaulting always returning them)
- If it should be limited only to `sharedWithUserId` (instead of just ignoring this aspect)

Furthermore, we already need a very similar query doing the same thing (`checkDuplicate()`) so it would make sense to re-use that here for this task, instead of using the already complex aggregate query.

So I think a separate API endpoint might still make sense for simplicity's sake, even if API paths point-of-view is different.

Not super strong opinion tho but just my trail of thought when I think about this. What do you think?

Reply:
@simison yeah, in general, I agree with your reasoning. Check out the new changes